go-controller/pkg/ovn/pods.go

```diff
@@ -122,8 +122,12 @@ func (oc *DefaultNetworkController) syncPods(pods []interface{}) error {
 		sw := nbdb.LogicalSwitch{
 			Name: switchName,
 		}
-		sw.UUID, _ = oc.lsManager.GetUUID(switchName)
-
+		var ok bool
```

Comment:
hmm... not having the sw UUID should be okay in this particular case, because the predicate would try finding it via name.

 @martinkennelly how about we instead of doing this, we did what was done here:
https://github.com/ovn-org/ovn-kubernetes/pull/3289#discussion_r1030390313
https://github.com/ovn-org/ovn-kubernetes/blob/d8976a3bda6ea8f222491faf4ba2c7b0922ee7bc/go-controller/pkg/ovn/pods.go#L271-L273

If so, what we should change instead is to make 
https://github.com/ovn-org/ovn-kubernetes/blob/2cceeebd4f66ee8dd9e683551b883e549b5cd7da/go-controller/pkg/libovsdbops/switch.go#L359  
use `%w`:
```
return nil, fmt.Errorf("error getting port %+v: %w", lsp, err)
```
and use that as:
```
if err != nil && !errors.Is(err, libovsdbclient.ErrNotFound) {
```
@kyrtapz wdyt?

Reply:
Flavio, I abandoned this approach because there are two sources within `DeleteLogicalSwitchPortsWithPredicateOps` that can generate `libovsdbclient.ErrNotFound`. The first is if the LS doesn't exist and the second is if it fails to lookup a LSP. A caller cannot decern which one failed and each one must be handled differently so I just decided to check if the LS exists before `DeleteLogicalSwitchPortsWithPredicateOps`.

Reply:
If a LS doesn't exist, i would have liked to continue, but if a port on that LS cannot be found, I didn't want to continue. As a caller to that func, I couldn't discern which failure occurred.

Reply:
ack. if nothing else I would like to have the test you created merged, so this codepath is exercised by ci.
like @jcaamano mentioned, I would like this code to remain idem-potent as possible. if port or ls is gone, the delete should be okay with it.